include/oneapi/mkl/dft/descriptor.hpp

```diff
@@ -51,22 +58,28 @@ class descriptor {
 
     void get_value(config_param param, ...);
 
-    void commit(sycl::queue& queue);
+    void commit(sycl::queue& queue) {
+        pimpl_.reset(detail::create_commit(get_device_id(queue), queue, values));
+    }
 
 #ifdef ENABLE_MKLCPU_BACKEND
-    void commit(backend_selector<backend::mklcpu> selector);
+    void commit(backend_selector<backend::mklcpu> selector) {
+        pimpl_.reset(mklcpu::create_commit(selector.get_queue(), values));
+    }
 #endif
 
 #ifdef ENABLE_MKLGPU_BACKEND
-    void commit(backend_selector<backend::mklgpu> selector);
+    void commit(backend_selector<backend::mklgpu> selector) {
+        // pimpl_.reset(mklgpu::create_commit<prec, dom>(selector.get_queue()));
+    }
 #endif
 
     sycl::queue& get_queue() {
         return queue_;
     }
 private:
     sycl::queue queue_;
```

Comment:
I think `queue_` should be in the `pimple_`. 

We don't know what queue the user wants until we commit, and default-constructing a queue that we don't plan on using seems like an odd choice.

Reply:
agreed, removed